### PR TITLE
Use _impact_ instead of _tel_, fixes #2051

### LINKED ISSUE #2051: reconstructed impact container has wrong prefix when loaded
[...] The Container has a prefix of "impact" so the intention was that the column name is: `impact_distance` not `tel_distance`. [...]

_Originally posted by @kosack in https://github.com/cta-observatory/ctapipe/issues/2048#issuecomment-1203837728_

### PR DESCRIPTION
**Pull request by nbiederbeck**

### CODE REVIEW

**Review by kosack — APPROVED** [review]

**Comment by nbiederbeck** [comment]
This currently breaks three other tests, two of which tests for `tel_distance`.

I'm trying to figure out why the other one fails...

**Review by kosack — CHANGES_REQUESTED** [review]
Need to update the data model version I guess, since this will break scripts reading files that were written with ctapipe-0.16 that assume the old naming (probably there are none, but still good to update).

**Comment by maxnoe** [comment]
The idea here was that all mono, telescope specific features have a tel_ prefix to distinguish them from array ones. This is important e.g. for joining tables in the table loader.

So please at least use `tel_impact`

**Comment by kosack** [comment]
> So please at least use tel_impact

Ok, agreed - I didn't like that the word `impact` got dropped somehow in the last refactoring, but having all telescope-wise parameters start with `tel_` is also very good.  So yes, let's please change it to "tel_impact_distance", which means just changing the prefix to "tel_impact" for that container.

**Comment by nbiederbeck** [comment]
Done. I think I changed it for all DL2 telescope-wise parameters, at least the tests don't tell me otherwise. I added the change to the DataModel description

**Review by kosack — APPROVED** [review]

File: ctapipe/io/datawriter.py [hunk 1]
```diff
@@ -46,8 +46,9 @@ def _get_tel_index(event, tel_id):
 #   (meaning readers need to update scripts)
 # - increase the minor number if new columns or datasets are added
 # - increase the patch number if there is a small bugfix to the model.
-DATA_MODEL_VERSION = "v4.0.0"
+DATA_MODEL_VERSION = "v5.0.0"
 DATA_MODEL_CHANGE_HISTORY = """
+- v5.0.0: - Change DL2 telescope-wise container prefixes from {algorithm}_tel to {algorithm}_tel_{kind}
```
Comment (maxnoe) [inline ctapipe/io/datawriter.py:51]: This indicates a much larger change than has actually happened.

We just renamed `tel_distance` to `tel_impact_distance`. I.e. whe changed the prefix of a single container from '' to 'impact'.

Nothing to do with "algorithms" or "kinds".

Comment (nbiederbeck) [reply ctapipe/io/datawriter.py:51]: Well yes, but in https://github.com/cta-observatory/ctapipe/blob/6cc4f66f6e61a8da2bbbc72507d66381adde91e0/ctapipe/io/hdf5eventsource.py#L449 I made the changes as indicated in the history. This currently only adds the `_impact_` to tel_distance, but (in future?) other DL2 telescope-wise parameters will have the "kind", in addition to the "algorithm", with the "tel" in between. I wanted to account for this in the history. 

But I understand, currently this changes only the `tel_distance`, and am fine with making that clearer

Comment (maxnoe) [reply ctapipe/io/datawriter.py:51]: Sounds good

**Review by kosack — APPROVED** [review]

**Comment by maxnoe** [comment]
Let's do a bugfix release first and try to sort out the test data before going to 0.17.

I'll approve, but let's wait with the merging...

**Comment by maxnoe** [comment]
On the other hand, this changes the data model (again) and there shouldn't be any data using 0.16 yet, so let's merge this and then reproduce test data.

**Review by maxnoe — APPROVED** [review]

**Review by maxnoe — APPROVED** [review]

**Pull request merged**